snapcraft/parts/setup_assets.py

```diff
@@ -37,46 +37,38 @@ def setup_assets(
     *,
     assets_dir: Path,
     project_dir: Path,
-    prime_dir: Path,
+    prime_dirs: dict[str | None, Path],
     meta_directory_handler: Optional[Callable[[Path, Path], None]] = None,
 ) -> None:
     """Copy assets to the appropriate locations in the snap filesystem.
 
     :param project: The snap project file.
     :param assets_dir: The directory containing snap project assets.
     :param project_dir: The project root directory.
-    :param prime_dir: The directory containing the content to be snapped.
+    :param prime_dirs: A mapping of component names to prime directories. 'None' should
+        map to the default prime directory.
     :param meta_directory_handler: callback that overrides default behavior for
         handling hooks and gui. The arguments passed are assets_dir and prime_dir.
     """
+    prime_dir = prime_dirs[None]
     meta_dir = prime_dir / "meta"
     gui_dir = meta_dir / "gui"
     gui_dir.mkdir(parents=True, exist_ok=True)
-    hooks_dir = meta_dir / "hooks"
-
-    if meta_directory_handler:
-        meta_directory_handler(assets_dir, prime_dir)
-    else:
-        _write_snap_directory(
-            assets_dir=assets_dir, prime_dir=prime_dir, meta_dir=meta_dir
-        )
-        # create wrappers for hooks in the snap/hooks directory
-        _create_hook_wrappers(prime_dir)
-
-    # hooks with command chains will execute the command chain instead of the hook wrapper
-    if project.hooks:
-        for hook_name, hook in project.hooks.items():
-            if hook.command_chain:
-                _validate_command_chain(
-                    hook.command_chain, name=f"hook {hook_name!r}", prime_dir=prime_dir
-                )
-            _ensure_hook(meta_dir / "hooks" / hook_name)
 
-    # Ensure all hooks are executable
-    if hooks_dir.is_dir():
-        for hook in hooks_dir.iterdir():  # type: ignore
-            # mypy says we are passing a Hook, but this is indeed a Path
-            _ensure_hook_executable(hook)  # type: ignore
+    copy_assets(assets_dir, prime_dir, meta_directory_handler)
+    setup_hooks(project.hooks, prime_dir)
+
+    # core22 doesn't provide a meta_directory_handler, which means that core22 component
+    # hooks would be handled like core22 snap hooks (with hook wrappers).
+    # This behavior is not desired, so component hooks are currently ignored for core22.
+    if project.components and project.get_effective_base() != "core22":
```

Comment:
This and an updated core22 spread test will give us core22 component hook support.

```suggestion
    if project.components:
```